client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/screen-category-list.tsx

```diff
@@ -0,0 +1,131 @@
+import { useSectionPatterns } from './patterns-data';
+import type { Pattern, Category } from './types';
+import './screen-category-list.scss';
+
+interface Props {
+	categories: Category[];
+	sectionsMapByCategory: { [ key: string ]: Pattern[] };
+	onDoneClick: () => void;
+	onSelect: (
+		type: string,
+		selectedPattern: Pattern | null,
+		selectedCategory: string | null
+	) => void;
+	replacePatternMode: boolean;
+	selectedPattern: Pattern | null;
+	wrapperRef: React.RefObject< HTMLDivElement > | null;
+}
+
+const ScreenCategoryList = ( {
+	sectionsMapByCategory,
+	categories,
+	onDoneClick,
+	replacePatternMode,
+	onSelect,
+	selectedPattern,
+	wrapperRef,
+}: Props ) => {
+	const translate = useTranslate();
+	const [ selectedCategory, setSelectedCategory ] = useState< string | null >( null );
+	const sectionPatterns = useSectionPatterns();
+	const categoriesInOrder = useCategoriesOrder( categories );
+
+	const handleFocusOutside = ( event: Event ) => {
+		// Click on large preview to close Pattern List
+		if ( ( event.target as HTMLElement ).closest( '.pattern-large-preview' ) ) {
+			setSelectedCategory( null );
+		}
+	};
+
+	useEffect( () => {
+		wrapperRef?.current?.addEventListener( 'click', handleFocusOutside );
+		return () => {
+			wrapperRef?.current?.removeEventListener( 'click', handleFocusOutside );
+		};
+	}, [] );
+
+	return (
+		<div className="screen-container">
+			<NavigatorHeader
+				title={ replacePatternMode ? translate( 'Replace pattern' ) : translate( 'Add patterns' ) }
+				description={
+					replacePatternMode
+						? translate(
+								'Replace the selected pattern by choosing from the list of categories below.'
+						  )
+						: translate(
+								'Find the right patterns for you by exploring the list of categories below.'
+						  )
+				}
+			/>
+			<div className="screen-container__body screen-category-list__body">
+				{ categoriesInOrder.map( ( { name, label, description } ) => {
+					const isOpen = selectedCategory === name;
+					const hasPatterns = sectionsMapByCategory[ name ]?.length;
+
+					if ( ! hasPatterns ) {
+						return null;
+					}
+
+					return (
+						<Button
+							key={ name }
+							className={ classNames( 'screen-category-list__category-button', {
+								'screen-category-list__category-button--is-open': isOpen,
+							} ) }
+							title={ description }
+							onClick={ () => {
+								if ( isOpen ) {
+									setSelectedCategory( null );
+								} else {
+									setSelectedCategory( name );
+								}
+							} }
+						>
+							<span>{ label }</span>
+							<Icon icon={ chevronRight } size={ 24 } />
+						</Button>
+					);
+				} ) }
+			</div>
+			<div className="screen-container__footer">
+				<NavigatorBackButton
+					as={ Button }
+					className="pattern-assembler__button"
+					onClick={ () => {
+						onDoneClick();
+					} }
+					primary
+				>
+					{ translate( 'Done' ) }
+				</NavigatorBackButton>
+			</div>
+			{ createPortal(
```

Comment:
I'll remove the portal and the CSS related to the animation because it's not required now that we don't have animation.